src/puzzles/day_7.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::{HashMap, HashSet};
+
+use regex::Regex;
+
+use super::input;
+
+pub fn part_1() -> i32 {
+    let input = input::read_lines(7);
+    let contains_rules = parse_rules(input.iter().map(|line| line.as_str()).collect());
+    let mut contained_in_rules: HashMap<&str, HashSet<&str>> = HashMap::new();
+
+    // Invert rules.
+    for (container_color, contains_rule) in contains_rules {
+        // println!("{:?} contains", container_color);
+        for (contained_color, _) in contains_rule {
+            // println!("\t{:?}", contained_color);
+            match contained_in_rules.get_mut(contained_color) {
+                Some(contained_in_rule) => {
+                    contained_in_rule.insert(container_color);
+                }
+                None => {
+                    let mut contained_in_rule = HashSet::new();
+                    contained_in_rule.insert(container_color);
+                    contained_in_rules.insert(contained_color, contained_in_rule);
+                }
+            }
+        }
+    }
+
+    println!("{:#?}", contained_in_rules);
+
+    let containers = get_containers("shiny gold", &contained_in_rules);
+
+    // println!("{:#?}", containers);
+
+    containers.len() as i32
+}
+
+pub fn part_2() -> i32 {
+    let input = input::read_lines(7);
+    let rules = parse_rules(input.iter().map(|line| line.as_str()).collect());
+    get_content_count("shiny gold", &rules)
+}
+
+fn get_containers<'a>(
```

Comment:
Add a function `get_containers_count` to wrap the `len()` call on the `HashSet`.